concierge_scheduler/concierge_cloud.py

```diff
@@ -0,0 +1,90 @@
+import json
+import os
+import sys
+import logging
+from google.cloud import storage
+from google.oauth2 import service_account
+
+
+class CloudBackupInterface:
+    def __init__(self, bucket_name, credential_file, config_dir, bucket_folder=''):
```

Comment:
The current usage of this class in `concierge_scheduler.py`: 
```python
    ...
    event_admin(zbx_client, cmd_args.config_dir, cmd_args.force_templates).run(cmd_args.command)
elif cmd_args.command in ['upload']: 
    cloud_admin(cmd_args.bucket_name, credential_files[cmd_args.cloud_engine], cmd_args.config_dir, 
       cmd_args.bucket_folder).run(cmd_args.command) 
```
Where `cloud_admin` is the class `GCSAdmin` (based on user input) and the run command runs `upload_local_dir()`.

I feel that all the arguments would be required for all cloud implementations:
* `bucket_name` - As far as I can see only Azure uses `blob` instead of `bucket`, but this variable would be required for all cloud storage functionality
* `credential_file` - This could be changed to `storage_client` which would be the authenticated client object, but since authentication happens within the same class as the other methods, it seems worth authenticating on class init. 
* `config_dir` - Path to directory where the configuration files for `upload` are locate. This would also apply for a download if that was implemented, as the directory to save the files to.
* `bucket_folder` - This is optional, so moved to only `GCSAdmin` class, but possibly still a useful feature across different cloud providers, as this is the folder within the bucket that the `config_dir` will be uploaded to. So files are `<bucket>/<bucket_folder>/<config_dir name>/<file>`